### PR TITLE
Add Windows CI support with AppVeyor

### PR DESCRIPTION
**Pull request by SethMichaelLarson**
Recently @Lukasa has added Mac OS support using Travis in order to fully test the new selectors addition.  One of the main projects that requires decent selectors support is the Happy Eyeballs algorithm (RFC 6555) but the algorithm also requires the use of Windows error constants in order to correctly work on all platforms.

I decided to go ahead and use a good template for testing Python with AppVeyor found at: https://github.com/ogrisel/python-appveyor-demo  I've changed it from a primarily Cython-focused project template to a tox-focused project and I've gotten AppVeyor to run all our tests on all our supported Python versions.  This will give us a lot less guessing when implementing or using features that are for platforms that are not available to all maintainers.  I have not tested the codecov section of the code, but I'm sure it's not too far off if at all.  I've already got all tests running. I'm sure this will take a repository owner to fully integrate AppVeyor support.

From my own trial runs, there are a few tests that fail on Windows already, especially in Python 2.6 with many TLS/SSL errors, there is also another test failing on all Python versions.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+environment:
+  matrix:
+    - PYTHON: "C:\\Python266"
+      PYTHON_VERSION: "2.6.6"
+      PYTHON_ARCH: "32"
+      TOXENV: "py26"
+
+    - PYTHON: "C:\\Python266-x64"
+      PYTHON_VERSION: "2.6.6"
+      PYTHON_ARCH: "64"
```
Comment (Lukasa) [inline appveyor.yml]: Do we have any reason to believe that Python architecture is going to be a relevant factor in bugs?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  - "pip install tox"
+
+  - "python setup.py install"
+
+test_script:
+  - "tox"
+
+on_success:
+  - "pip install codecov"
+  - "codecov --env PLATFORM,TOXENV"
```
Comment (Lukasa) [inline appveyor.yml]: This needs to run inside the `tox` environment to ensure that it gets the same coverage version as `tox` installs. See [this script for Travis](https://github.com/shazow/urllib3/blob/master/_travis/upload_coverage.sh) that does exactly that.

File: appveyor.yml (hunk 1 above)
Comment (SethMichaelLarson) [inline appveyor.yml]: Probably not, was just a part of the script. I'll use all 64 bit then?

File: appveyor.yml (hunk 2 above)
Comment (SethMichaelLarson) [inline appveyor.yml]: Sure, I can do that.

**Comment by SethMichaelLarson** [comment]
Made the necessary changes, ready for re-review @Lukasa :)

**Comment by Lukasa** [comment]
Ok @shazow, this one will require your intervention. You'll need to sign up for Appveyor and point this project at it, before we merge this patch. I should note that this patch will *immediately* fail on Appveyor, but that's ok: we'll start another pull request to fix it.

**Comment by SethMichaelLarson** [comment]
Noting here that #1000 is dependent on this PR for complete coverage as it uses Windows-specific error numbers.

**Comment by SethMichaelLarson** [comment]
Quick ping, would like to have this support while working on #1000 :)

**Comment by SethMichaelLarson** [comment]
This would be very nice to have with all the lower-level stuff we're doing and I'm sure not everyone has a Windows machine to test things on. :)

**Comment by Lukasa** [comment]
Yup, I'm very much still aware of this. :) Just gotta make sure I find a time where I overlap with @shazow to get him to hook up the relevant stuff.

**Comment by SethMichaelLarson** [comment]
Sorry @Lukasa :P I might as well be a cron job running `0 7 * * 1 "Ping Github"` :)

**Comment by SethMichaelLarson** [comment]
@Lukasa Could this be merged in before shazow logs into AppVeyor? It would allow me to work on fixing the tests from my own AppVeyor instance before we want to start actually running tests on AppVeyor. :)

**Comment by Lukasa** [comment]
@SethMichaelLarson You ought to be able to do that anyway. You can make changes to the master branch on your own fork and run AppVeyor against that, I think.

**Comment by SethMichaelLarson** [comment]
@shazow Is there now an AppVeyor for urllib3?

**Comment by haikuginger** [comment]
@SethMichaelLarson, [it would appear so!](https://ci.appveyor.com/project/shazow/urllib3) Thanks for rocking it on this PR! ✨🍪✨

**Comment by SethMichaelLarson** [comment]
Might want to get something to run correctly on the AppVeyor. I suspect that this actually may fail.

**Comment by SethMichaelLarson** [comment]
@haikuginger I'm addressing the issues that are currently in the AppVeyor tests.

**Comment by Lukasa** [comment]
Yeah, that's why we needed AppVeyor. Now we can get it to pass, and then work on some other stuff. Woo!

**Comment by SethMichaelLarson** [comment]
Submitted a new PR with hopefully all the changes we need to fix up tests for Windows.

**Comment by shazow** [comment]
@SethMichaelLarson Indeed, thank you! :) Let me know if you need me to make any other configuration changes on my end.

**Comment by SethMichaelLarson** [comment]
For now it looks good! :) Thanks!

**Pull request merged**